help: Fix "Typing notifications" intro paragraph.

drrosa opened the pull request:
- Corrects documentation to reflect that [typing notifications](https://zulip.com/help/typing-notifications) work when narrowing to all direct messages as well, adding a link to a new section in [Direct messages](https://zulip.com/help/direct-messages).

**Screenshots and screen captures:**
![image](https://github.com/zulip/zulip/assets/2343554/36a5dd90-f326-4a42-94c0-9dec261538d6)

![view-all-dms-relative-link](https://github.com/zulip/zulip/assets/2343554/d57adb99-682a-46eb-a123-3d8d3b6d35fe)
![view-all-dms-instructions](https://github.com/zulip/zulip/assets/2343554/fbb282e9-d761-4409-93d1-25d6a621cdbe)
![image](https://github.com/zulip/zulip/assets/2343554/5965e1cb-9355-4b5b-8538-72af4f45584f)

<details>
<summary>Self-review checklist</summary>

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Automated tests.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Links.
</details>

drrosa commented:
@laurynmm This PR is ready for review.
@zulipbot add "help center review"

laurynmm commented:
@drrosa - Does the mobile app support typing notifications for DMs? If so, I wonder if we should think about specifying that they are visible in the all direct messages view for the web app in that sentence, because I don't imagine that's the case in the mobile app as the all DM view is a list of the users in the conversations vs a message list view.

I like the added section to the "Direct messages" article. And imagine you've got adding mobile instructions to the two sections (**Access a DM or group DM** and **View all direct messages**) as a follow-up / future task.

drrosa commented:
> Does the mobile app support typing notifications for DMs? If so, I wonder if we should think about specifying that they are visible in the all direct messages view for the web app in that sentence, because I don't imagine that's the case in the mobile app as the all DM view is a list of the users in the conversations vs a message list view.

Yeah, I think it makes sense to specify that typing notifications are also displayed in the web/desktop app when a user is typing from the mobile app. Also, I noticed that typing notifications disappear if typing is paused for about 8 seconds rather than 15.

Does the following look good to you?

![image](https://github.com/zulip/zulip/assets/2343554/27978e81-7bf2-40c7-9aa3-5a61123d5856)

> I like the added section to the "Direct messages" article. And imagine you've got adding mobile instructions to the two sections (**Access a DM or group DM** and **View all direct messages**) as a follow-up / future task.

Thanks! Yes, these will be future tasks.

laurynmm commented:
@drrosa - So, I checked yesterday and there are typing notifications displayed (three animated dots) in the mobile app. Therefore, the above isn't quite right.

I'd go with your original sentence addition, but specify that the all direct messages typing notification is in the web/desktop apps only. Because the all direct messages view in the web app doesn't show messages, but rather just the people in the conversation.

Also, the 8 seconds seems odd, as it's set to 15 seconds in the frontend code, [CODE LINK](https://github.com/zulip/zulip/blob/2db715030d86819de8493afb879fb5c9c5e43139/web/src/typing_events.js#L18-L20). And it's documented as 15 seconds in the API docs, see [Set "typing" status](https://zulip.com/api/set-typing-status) endpoint doc. I'd check in CZO before we change that information, because if the time is really less then it might be a regression?

drrosa commented:
> @drrosa - So, I checked yesterday and there are typing notifications displayed (three animated dots) in the mobile app. Therefore, the above isn't quite right.

@laurynmm Hm, I also checked yesterday on both iOS and Android and didn't see three animated dots in the mobile app. Were you  checking the latest release or a beta release? Maybe I'm missing something, can you post a screenshot of the typing notifications in the mobile app?

> Also, the 8 seconds seems odd, as it's set to 15 seconds in the frontend code, [CODE LINK](https://github.com/zulip/zulip/blob/2db715030d86819de8493afb879fb5c9c5e43139/web/src/typing_events.js#L18-L20). And it's documented as 15 seconds in the API docs, see [Set "typing" status](https://zulip.com/api/set-typing-status) endpoint doc. I'd check in CZO before we change that information, because if the time is really less then it might be a regression?

Yeah, that's odd.  I'll start a thread about this in CZO. I'm curious whether it's 8 or 15 seconds on your end, can you check?

laurynmm commented:
I don't think you need to start a thread on CZO about the timing.

Looking at the code, 15 seconds is the max that it would be there. E.g. if the person typing had their computer shut down suddenly or the browser crashed, and the server never got a message saying this person stopped typing to you, then person receiving the notification would have that notification only be there for 15 seconds max before timing out on their end (aka getting nothing from the server re:stop or start typing notifications).

If the person stops typing for 5 seconds, then they send a notification to the server that they stopped typing. So really, it's a 5-15 second period with network speed and all of that layered on.

So maybe we say: "and they disappear if typing is paused for a bit."? Or "paused for 5 seconds or more."? Or "paused for a few moments"?

drrosa commented:
> Hm, I also checked yesterday on both iOS and Android and didn't see three animated dots in the mobile app. 

@laurynmm So it turns out that typing notifications were not being displayed in the mobile app because I was testing by typing the very first message in that DM conversation (typing notifications do work in this case in the web/desktop app). Once I sent the first message, the typing notifications in the mobile app became visible for subsequent messages. :)

Thanks for the feedback! I updated the PR and screenshot in the PR description.

laurynmm commented on help/typing-notifications.md:
```diff
@@ -2,12 +2,16 @@
+participants is currently composing a message. Typing notifications
+are also displayed in the
+[all direct messages view](/help/direct-messages#view-all-direct-messages)
+for the Zulip web and desktop apps only.
```
How about:

> In the web and desktop apps, typing notifications are also displayed in the [all direct messages view](/help/direct-messages#view-all-direct-messages).

laurynmm commented on help/typing-notifications.md:
```diff
@@ -2,12 +2,16 @@
+Typing notifications are only sent while one is actively editing text in
+the compose box, and they disappear if typing is paused for at least 5
+seconds, if all the content of the message is erased, or if the message is
+[saved as a draft](/help/view-and-edit-your-message-drafts#save-a-draft).
```
I think I'd make these two separate sentences now ...

> Typing notifications are only sent while one is actively editing text in the compose box. They disappear if typing is paused ...

laurynmm reviewed (commented):
@drrosa - I think the additional details about when the notifications stop is great. Glad we figured out the timing and the mobile app different.

I had two small revisions, but overall looks good!

drrosa commented:
@laurynmm I also went ahead and added mobile instructions for accessing all DMs, and changed the subheading to be consistent with the other sections in the article. What do you think?

![image](https://github.com/zulip/zulip/assets/2343554/32a90cac-0b5d-47af-8fa0-947ea0f85000)

timabbott commented on help/typing-notifications.md:
```diff
@@ -2,12 +2,15 @@
+
+Typing notifications are only sent while one is actively editing text in
+the compose box. They disappear if typing is paused for at least 5
+seconds, if all the content of the message is erased, or if the message is
```
It might be better to just say "several" -- I am not sure this correctly summarizes the algorithm, and in any case it's probably more resolution than should be relevant to users.

drrosa replied:
Yeah, that makes sense. Done!

laurynmm commented:
@drrosa - Looks good to me! Marking for integration review.

timabbott commented:
Merged, thanks @drrosa and @laurynmm!
